GitCommands/RevisionReader.cs

```diff
@@ -395,19 +395,23 @@ long ParseUnixDateTime(ref ReadOnlySpan<byte> array)
             var committer = reader.ReadLine();
             var committerEmail = reader.ReadLine();
 
-            // We keep a full multiline message body within the last six months.
-            // Note also that if body and subject are identical (single line), the subject never need to be stored
-            bool skipBody = sixMonths > authorUnixTime;
-            (string? subject, string? body, bool hasMultiLineMessage) = reader.ReadToEndOfBody(skipBody, hasPathFilter);
+            (string? subject, string? body, bool hasMultiLineMessage) = reader.ReadToEndOfBody(hasPathFilter);
 
-            if (author is null || authorEmail is null || committer is null || committerEmail is null || subject is null || (!skipBody && hasMultiLineMessage && body is null))
+            if (author is null || authorEmail is null || committer is null || committerEmail is null || subject is null || body is null)
             {
                 // TODO log this parse error
                 Debug.Fail("Unable to read an entry from the log -- this should not happen");
                 revision = default;
                 return false;
             }
 
+            // We keep a full multiline message body within the last six months.
+            // Note also that if body and subject are identical (single line), the subject never need to be stored
+            if (sixMonths > authorUnixTime || !hasMultiLineMessage)
+            {
+                body = null;
+            }
```

Comment:
Why have you switched back to GC'ing the `body`?

Reply:
It This was to simplify, but I will revert so my previous version...